compiler/rustc_typeck/src/check/check.rs

```diff
@@ -1375,7 +1375,10 @@ fn check_enum<'tcx>(
         }
     }
 
+
     let mut disr_vals: Vec<Discr<'tcx>> = Vec::with_capacity(vs.len());
+    // This tracks the previous variant span (in the loop) incase we need it for diagnostics
+    let mut prev_variant_span: Span = Span::default();
```

Comment:
`DUMMY_SP` is the dummy span -- we could use an `Option` here but it's overkill.